### PR TITLE
Fallback to upload current dir when running cvdr create with local_image

### PR DESCRIPTION
**Pull request by 0405ysj**
Testing method for the ARM server:
```
cvdr \
--http_proxy=socks5://localhost:1337 \
--service_url=http://100.114.242.57:8080 \
--zone=local \
--host=$CO_HOST_NAME \
--local_image=true \
--auto_connect=false \
create
```

### CODE REVIEW

**Comment by 0405ysj** [comment]
Interested thing was, it took only about 5-7 mins to upload artifacts & running CF instance. Probably, would it be caused by using public stun server(stun.l.google.com)?

**Comment by ikicha** [comment]
Interesting, I think it might be due to socks5 proxy rather than stun server. can you benchmark network bandwidth with socks5?

**Comment by 0405ysj** [comment]
I roughly checked with our hybrid one, and it took about 12-13 mins. And `super.img` was about 4.7G. Seems like the bandwidth is about 6MB/s or 48Mbps

File: pkg/cli/cvd.go [hunk 1]
```diff
@@ -143,17 +143,30 @@ func (c *cvdCreator) Create() ([]*hoapi.CVD, error) {
...
-		return nil, fmt.Errorf("error retrieving Android Build environment variables: %w", err)
-	}
-	names, err := ListLocalImageRequiredFiles(vars)
-	if err != nil {
-		return nil, fmt.Errorf("error building list of required image files: %w", err)
-	}
-	if err := verifyCVDHostPackageTar(vars.HostOut); err != nil {
-		return nil, fmt.Errorf("invalid cvd host package: %w", err)
+	var names []string
+	if err == nil {
```
Comment (jemoreira) [inline pkg/cli/cvd.go:147]: The code pattern for this function should be roughly:
```
cvd, err := createCVDFromBuiltSource(...)
if errors.Is(err, NotInAndroidSourceErr) {
  return createCVDFromCurrentDir(...)
}
return cvd, err
```
I think encapsulating both flows into their own functions will increase readability.

**Review by jemoreira — CHANGES_REQUESTED** [review]

**Comment by ser-io** [comment]
Uploading everything from the current directory with no validation is a big responsibility to pass to the final user, this could end up in undesirable behaviors, for example uploading personal files.

**Comment by ikicha** [comment]
IIUC, imgs.zip has required_image file inside. We can use that to list up files which we want to send.

**Comment by 0405ysj** [comment]
> Wrote https://github.com/google/cloud-android-orchestration/pull/292 where we need to be upfront with what's going to be uploaded, it also offers more flexibility to build instances with mixed artifacts, as we just need to change the artifacts sources file.

Should I close this PR then? I saw the PR you've created, and this and that contains duplicated contents. Seems your PR covers wider purposes.


> IIUC, imgs.zip has required_image file inside. 

Do we? I unpacked CF image zip file downloaded from ci.android.com, and there was no `required_image`.

**Comment by ikicha** [comment]
ah, sorry, require_image exists in PRODUCT_OUT, not imgs.zip

**Comment by ser-io** [comment]
yes, let's close this one, thanks!

**Pull request closed without merging**